src/Avalonia.Controls/Platform/PlatformManager.cs

```diff
@@ -25,9 +25,7 @@ public static void SetDesignerScalingFactor(double factor)
             var platform = AvaloniaLocator.Current.GetService<IWindowingPlatform>();
 
             if (platform == null)
-            {
-                throw new Exception("Could not CreateTrayIcon(): IWindowingPlatform is not registered.");
-            }
+                return null;
 
             return s_designerMode ? null : platform.CreateTrayIcon();
```

Comment:
Maybe simplify this to `return s_designerMode ? null : platform?.CreateTrayIcon();` and remove the `if`?